enumerable.rb

```diff
@@ -0,0 +1,177 @@
+      n.times do |i|
+        yield(v[i])
+      end
+    else
+      to_enum(:my_each)
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      v = self
+      n = v.length
+      n.times do |i|
+        yield(self[i], i)
+      end
+    else
+      to_enum(:my_each_with_index)
+    end
+  end
+
+  def my_select
+    value = self
+    if block_given?
+      temp = []
+      value.my_each do |a|
+        temp.push(a) if yield(a)
+      end
+      temp
+    else
+      to_enum(:my_select)
+    end
+  end
+
+  def my_all?(arg = nil)
+    value = self
+    if block_given?
+      if value.is_a? Hash
+        value.my_each { |a, b| return false unless yield(a, b) }
+      else
+        value.my_each { |a| return false unless yield(a) }
+      end
+    elsif arg.is_a? Regexp
+      value.my_each { |a| return false unless a =~ arg }
+    elsif arg.is_a? Class
+      value.my_each { |a| return false unless a.class == arg }
+    elsif arg.nil?
+      value.my_each { |a| return false unless a }
+    else
+      value.my_each { |a| return false unless a == arg }
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    value = self
+    if block_given?
+      if value.is_a? Hash
+        value.my_each { |a, b| return true if yield(a, b) }
+      else
+        value.my_each { |a| return true if yield(a) }
+      end
+    elsif arg.is_a? Regexp
+      value.my_each { |a| return true if a =~ arg }
+    elsif arg.is_a? Class
+      value.my_each { |a| return true if a.class == arg }
+    elsif arg.nil?
+      value.my_each { |a| return true if a }
+    else
+      value.my_each { |a| return true if a == arg }
+    end
+    false
+  end
+
+  def my_none?(arg = nil)
+    value = self
+    if block_given?
+      if value.is_a? Hash
+        value.my_each { |a, b| return false if yield(a, b) }
+      else
+        value.my_each { |a| return false if yield(a) }
+      end
+    elsif arg.is_a? Regexp
+      value.my_each { |a| return false if a =~ arg }
+    elsif arg.is_a? Class
+      value.my_each { |a| return false if a.class == arg }
+    elsif arg.nil?
+      value.my_each { |a| return false if a }
+    else
+      value.my_each { |a| return false if a == arg }
+    end
+    true
+  end
+
+  def my_count(arg = nil)
+    value = self
+    count = 0
+    if block_given?
+      value.my_each do |a|
+        count += 1 if yield(a)
+      end
+    end
+
+    unless arg.nil?
+      value.my_each do |a|
+        count += 1 if arg == a
+      end
+    end
+
+    count = value.length if !block_given? && arg.nil?
+    count
+  end
+
+  def my_map(&block)
+    arr = []
+    value = self
+    if block_given?
+      if value.is_a? Hash
+        value.my_each { |a, b| arr.push(block.call(a, b)) }
+      elsif value.is_a? Array
+        value.my_each { |a| arr.push(block.call(a)) }
+      end
+      arr
+    else
+      to_enum(:my_map)
+    end
+  end
+
+  def my_inject(init = nil, arg = nil, &block)
+    value = self
+    value = value.to_a
```

Comment:
**[REQUIRED]**
Let's change ` value = value.to_a` to `value = value.dup.to_a`.

The original array on which the method is called should not be mutated/changed.
`shift` method mutates the original array thus  causing problems.


This change will fix the error below :point_down:  
![he](https://user-images.githubusercontent.com/44072711/68615539-473a6600-0491-11ea-94c7-e4cf4a4e9ef7.png)